### PR TITLE
Include Health Check Indicators for application status monitoring

### PR DESCRIPTION
**Pull request by ewelinagr**
Add `/health` endpoint to monitor availability of the application.

Related ticket: [TMT-944]( https://jira.thehyve.nl/browse/TMT-944)

### CODE REVIEW

File: open-api/swagger.yaml [hunk 1]
```diff
@@ -588,6 +588,20 @@ paths:
                 $ref: '#/components/schemas/Version'
         '404':
           description: Version not available.
+  /health:
```
Comment (forus) [inline open-api/swagger.yaml]: As it's spring boot functionality that we document here I recommend mention that in the description and add a link to the official documentation.

File: transmart-api-server/grails-app/conf/application.yml [hunk 2]
```diff
@@ -23,6 +23,14 @@ endpoints:
     enabled: false
     jmx:
         enabled: true
+    health:
```
Comment (forus) [inline transmart-api-server/grails-app/conf/application.yml]: It looks like `management.security.enabled` and `endpoints.health.sensitive` detect what behavior to be for unauthenticated or authenticated users https://docs.spring.io/spring-boot/docs/1.5.3.RELEASE/reference/html/production-ready-monitoring.html#production-ready-health-access-restrictions
I assume that now we have:
`management.security.enabled=true`
`endpoints.health.sensitive=false`
Maybe it makes sense to make it explicit here what state of flags are and why it's important.

File: open-api/swagger.yaml (hunk 1 above)
Comment (ewelinagr) [reply open-api/swagger.yaml]: good idea!

File: transmart-api-server/grails-app/conf/application.yml (hunk 2 above)
Comment (ewelinagr) [reply transmart-api-server/grails-app/conf/application.yml]: there is some issue with `endpoints.health.sensitive`, changing it does not affect the behavior at all, so I would rather not add it here. But regarding `endpoints.health.sensitive` you're right. Apparently the defaults differ between spring-boot-actuator version, at least according to the documentation. I'll add it here.

**Pull request merged**